### PR TITLE
CBAPI-3435: Added info for async export with jobs

### PR DESCRIPTION
**Pull request by abowersox-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Tests have been added that prove the fix is effective or that the feature works.
- [ ] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?

https://jira.carbonblack.local/browse/CBAPI-3435

## Pull Request Description

* Linked "jobs" module contents onto Platform API page.
* Added more information to the docstring for `ResultQuery.async_export()`, mentioning what to do with the Job object once the caller gets it.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

New docs viewed locally to ensure that changes are reflected in generated documentation.

### CODE REVIEW

File: src/cbc_sdk/audit_remediation/base.py [hunk 1]
```diff
@@ -1413,7 +1413,9 @@ def async_export(self):
         """
         Create an asynchronous job that exports the results from the run.
 
-        This is recommended if you are expecting a very large result set.
+        This is recommended if you are expecting a very large result set.  Once the Job is created, wait for it to be
+        completed (using its await_completion() method or similar), then get the results from the Job using one of the
```
Comment (avanbrunt-cb) [inline src/cbc_sdk/audit_remediation/base.py:1417]: What is `or similar`? I would expand this to indicate that you can write your own logic to wait for the state to be marked completed. Preferably adding a snippet of sample code

Comment (abowersox-cb) [reply src/cbc_sdk/audit_remediation/base.py:1417]: I'll expand that to say `or some other method that waits for the status of the job to be completed`, but let's not clutter up this docstring by adding in sample code for something that doesn't even directly relate to `async_export()`.

Comment (avanbrunt-cb) [reply src/cbc_sdk/audit_remediation/base.py]: `some other method` - I'd ask what other functions can I use to wait? Really what you a recommending is that they write their own logic to wait for the job completion. It's valuable to reference the helper function but I would do just that as a recommendation. What about adding that after the sentence and embedding a link to that section in the text

```
        This is recommended if you are expecting a very large result set.  Once the Job is created, wait for it to be
        completed, then get the results from the Job using one of the get_output_ methods on the Job object. If need to 
        wait for the results asynchronously use the [await_completion()](link....) method in the Job object.
```

**Review by avanbrunt-cb — APPROVED** [review]

File: src/cbc_sdk/audit_remediation/base.py (hunk 1 above)
Comment (abowersox-cb) [reply src/cbc_sdk/audit_remediation/base.py]: OK, I rephrased it again.

File: src/cbc_sdk/audit_remediation/base.py [hunk 2]
```diff
@@ -1413,7 +1413,9 @@ def async_export(self):
         """
         Create an asynchronous job that exports the results from the run.
 
-        This is recommended if you are expecting a very large result set.
+        This is recommended if you are expecting a very large result set.  Once the Job is created, wait for it to be
+        completed, then get the results from the Job using one of the get_output_ methods on the Job object.  To wait
```
[the remark this answers is not in the document]
Comment (abowersox-cb) [reply src/cbc_sdk/audit_remediation/base.py:1417]: It wasn't intended to be a link at all. How can I keep it from being so?

[the remark this answers is not in the document]
Comment (kebringer-cb) [reply src/cbc_sdk/audit_remediation/base.py:1417]: removing the trailing underscore stops get_output being a link, and this is is the syntax to have a link to Job class.

...Job using one of the get_output methods on the :py:meth:`cbc_sdk.platform.jobs` object...

File: src/cbc_sdk/audit_remediation/base.py [hunk 3]
```diff
@@ -1413,7 +1413,9 @@ def async_export(self):
...
+        asynchronously for the results, use the Job object's await_completion() method.
```
[the remark this answers is not in the document]
Comment (kebringer-cb) [reply src/cbc_sdk/audit_remediation/base.py:1418]: in that should case async_export() have input parameters?  There's no info currently on the async_export method about how to change the output format.

File: src/cbc_sdk/audit_remediation/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (abowersox-cb) [reply src/cbc_sdk/audit_remediation/base.py:1417]: Replacing that part of the text with your example.

File: src/cbc_sdk/audit_remediation/base.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (abowersox-cb) [reply src/cbc_sdk/audit_remediation/base.py:1418]: That's because you _can't_ change the output format. The API doesn't allow it.

**Review by lhilmes-cb — APPROVED** [review]
Looks good!

**Review by kebringer-cb — APPROVED** [review]

**Pull request merged**